modelscope/hub/modelscope_mcp/api.py

```diff
@@ -0,0 +1,407 @@
+# Copyright (c) Alibaba, Inc. and its affiliates.
+"""
+MCP (Model Context Protocol) API interface for ModelScope Hub.
+
+This module provides API interfaces for interacting with MCP servers
+through the ModelScope Hub, including server management and deployment.
+"""
+
+import logging
+from typing import Any, Dict, Optional
+
+import requests
+
+from modelscope.hub.api import HubApi
+from modelscope.hub.errors import raise_for_http_status
+
+# Configure logging
+logger = logging.getLogger(__name__)
+
+# MCP API path suffix
+MCP_API_PATH = '/openapi/v1'
+
+
+class MCPApiError(Exception):
+    """Base exception for MCP API errors."""
+    pass
+
+
+class MCPApiRequestError(MCPApiError):
+    """Exception raised when MCP API request fails."""
+    pass
+
+
+class MCPApiResponseError(MCPApiError):
+    """Exception raised when MCP API response is invalid."""
+    pass
+
+
+class McpApi(HubApi):
+    """
+    MCP (Model Context Protocol) API interface class.
+
+    This class provides methods for interacting with MCP servers through
+    the ModelScope Hub API, including listing, deploying, and managing servers.
+    """
+
+    def __init__(self, base_api: HubApi) -> None:
+        """
+        Initialize MCP API.
+
+        Args:
+            base_api: HubApi instance for accessing basic API functionality
+
+        Raises:
+            ValueError: If base_api is not a valid HubApi instance
+        """
+
+        super().__init__(base_api)
+        # Inherit HubApi's endpoint but add OpenAPI-specific path
+        self.endpoint = base_api.endpoint + MCP_API_PATH
+        self.session = base_api.session
+        self.builder_headers = base_api.builder_headers
+        self.headers = base_api.headers
+
+    def get_mcp_servers(self,
+                        token: Optional[str] = None,
+                        filter_dict: Optional[Dict[str, Any]] = None,
+                        page_number: int = 1,
+                        page_size: int = 20,
+                        search: str = '',
+                        endpoint: Optional[str] = None) -> Dict[str, Any]:
+        """
+        Get MCP server list.
+
+        Args:
+            token: Authentication token (optional). If not provided, request will be made without authentication
+            filter_dict: Filter condition dictionary containing the following sub-branches:
+                - category: String type, filter by category
+                - is_hosted: Boolean type, filter by hosting status
+                - tag: JSON string type, filter by tags
+                When category, is_hosted, and tag are all provided, take the intersection of all three
```

Comment:
if multiple filters are provided, the server will return results satisfying all filetering conditions.